### PR TITLE
Updated SQL queries

### LINKED ISSUE #7: Update EntryDao to match the database column names

### PR DESCRIPTION
**Pull request by jayknightt**
- Added SQL queries 
- Implemented `searchEntriesByCriteria` method to search entries based on description, vendor, minAmount, and maxAmount
- Built dynamic SQL query with conditional parameters for flexible searching
- Mapped query results to `Entry` objects and returned as a list
- Handled potential SQL exceptions

### CODE REVIEW

File: src/main/java/org/yearup/data/mysql/MySqlEntryDao.java [hunk 1]
```diff
@@ -131,6 +131,55 @@ public void deleteEntry(int entryId)
         }
     }
 
+    //Custom Query Search
```
Comment (sekwanaa) [inline src/main/java/org/yearup/data/mysql/MySqlEntryDao.java]: Hey Jaleel, this looks pretty good! Could you actually move this code into the top method, that says searchEntries()?

This way, we don't have two methods trying to do the same thing.

File: src/main/java/org/yearup/data/mysql/MySqlEntryDao.java [hunk 2]
```diff
@@ -131,6 +131,55 @@ public void deleteEntry(int entryId)
...
+    public List<Entry> searchEntriesByCriteria(String description, String vendor, BigDecimal minAmount, BigDecimal maxAmount)
+    {
+        List<Entry> entries = new ArrayList<>();
+        StringBuilder query = new StringBuilder("SELECT * FROM entries WHERE 1=1");
```
Comment (sekwanaa) [inline src/main/java/org/yearup/data/mysql/MySqlEntryDao.java]: Also, if you reference the database, the column name is not 'entries' it is 'entry' so please make the necessary changes to all of the methods.

**Review by sekwanaa — CHANGES_REQUESTED** [review]
Hey Jaleel, overall looks good, just a couple small tweaks I need you to make and it should be good!

File: src/main/java/org/yearup/data/mysql/MySqlEntryDao.java (hunk 1 above)
Comment (jayknightt) [reply src/main/java/org/yearup/data/mysql/MySqlEntryDao.java]: Gotcha, ill get right on it

File: src/main/java/org/yearup/data/mysql/MySqlEntryDao.java (hunk 2 above)
Comment (sekwanaa) [reply src/main/java/org/yearup/data/mysql/MySqlEntryDao.java]: Haha, hey Jaleel, I noticed you just changed the List<Entry> entries to List<Entry> entry.

I meant more along the lines of in your sql query, you're calling SELECT * FROM entries, but the column name in the database is 'entry'.

File: src/main/java/org/yearup/data/mysql/MySqlEntryDao.java [hunk 3]
```diff
@@ -22,26 +22,66 @@ public MySqlEntryDao(DataSource dataSource)
     }
 
     @Override
-    public List<Entry> searchEntries()
+    public List<Entry> searchEntry(String description, String vendor, BigDecimal minAmount, BigDecimal maxAmount)
     {
-        List<Entry> entries = new ArrayList<>();
-        String query = "SELECT * FROM entries";
+        List<Entry> entry = new ArrayList<>();
```
Comment (sekwanaa) [inline src/main/java/org/yearup/data/mysql/MySqlEntryDao.java:27]: On this line, you changed entries to entry, but I was referring to the line below

**Review by sekwanaa — APPROVED** [review]
Great stuff

**Pull request merged**